activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -121,6 +121,12 @@ def self.define_validations(model, reflection)
         required = !reflection.options[:optional]
       end
 
+      if reflection.options[:polymorphic].is_a?(Array)
+        model.validates reflection.foreign_type,
+          inclusion: { in: reflection.options[:polymorphic].map { |klass| klass.to_s.freeze } },
```

Comment:
Why do we need to `freeze` here? Is it to dedup the string and not have a lot of allocations? If that is the case it should not matter, this only run once per call during boot time.

Reply:
Yeah I was trying to avoid string allocations by freezing them. These strings are retained and kept in the validator for the life of the app, right?

Reply:
Yes, but I don't think that matters

Reply:
Okay. I was trying to get these strings to share with `User.name`, since that's frozen. But I can just remove it. I was trying to think of what changes Jean might to this later, and do them first.

Reply:
I've removed the `.freeze`. I also added a sentence to the `belongs_to` API documentation, and rebased.